ios/NotificationServiceExtension/NotificationService.swift

```diff
@@ -8,12 +8,18 @@
 import AirshipServiceExtension
 import os.log
 import Intents
+import AppLogs
 
 class NotificationService: UANotificationServiceExtension {
   
   var contentHandler: ((UNNotificationContent) -> Void)?
   var bestAttemptContent: UNMutableNotificationContent?
   let log = OSLog(subsystem: Bundle.main.bundleIdentifier ?? "com.expensify.chat.dev.NotificationServiceExtension", category: "NotificationService")
+  let appLogs: AppLogs = .init()
+  
+  deinit {
+    appLogs.interceptLogs(appGroup: "group.com.expensify.new")
+  }
   
   override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
     os_log("[NotificationService] didReceive() - received notification", log: log)
```

Comment:
```suggestion
    os_log("NotificationService - didReceive() - received notification", log: log)
```

Can you update all log statements in the `NotificationService` like this? The end result should be that the logs get written to JS as `[PushNotification] NotificationService - ...`

Reply:
@arosiclair that's something I'd like to discuss with you before making any code changes.

The current code `log.message.replace(filter, '[PushNotification]');` is needed because in case if that message gets sent into `console.*` method, then it'll be automatically send to native logs and our handler would be fired again. Thus one single log without additional sanitization may produce an infinite loop of logs, i. e.:

- we receive "NotificationService - didReceive() - received notification"
- we somehow log it to `console.debug`/`console.log`/etc. as "[PushNotification] NotificationService - didReceive() - received notification"
- log from `console.debug`/`console.log`/etc. gets moved to native logger
- our handler sees a new log ("[PushNotification] NotificationService - didReceive() - received notification") that satisfies a condition (it includes a `NotificationService` substring) and pushes it to JS;
- we receive "[PushNotification] NotificationService - didReceive() - received notification" string and now we are in infinite loop 🙃 

I'm not against the idea of re-organizing log statements before sending them to the server, but just want to be sure that we are on the same page and we both understand the impact of such changes.

My idea is that it's better to remove the original string-filter (`NotificationService`) from the original log 🤷‍♂️

Reply:
> in case if that message gets sent into console.* method, then it'll be automatically send to native logs and our handler would be fired again

Hm I understand your concern but is this actually happening anywhere? IIRC, `console` is not allowed in the code base. So maybe that can be an issue for someone debugging with log statements locally (AND their log statements match the filter for some reason) but that's not something we need to avoid for this.

Reply:
And Log.info and Log.hmmm will never send their log to console, right?

Reply:
@arosiclair I just tested that behavior and what I got. I used this code in `App.tsx`:

```tsx
setTimeout(() => {
    Log.info("[NotificationService] tra-la-la");
}, 10000);
```

I. e. I simulate intercepting of the log from `NotificationService` and just log the message (without message modification).

And used this code for `react-native-app-logs`:

```tsx
AppLogs.configureAppGroupName('group.com.expensify.new');
AppLogs.registerHandler({
    filter: '[NotificationService]',
    handler: ({filter, logs}) => {
        logs.forEach((log) => {
            // sanitize filter to avoid infinite loop
            const message = log.message;
            if (log.level === 'error') {
                Log.hmmm(message);
            } else {
                Log.info(message);
            }
        });
    },
});
```

The first code (setTimeout) produces:

```
 DEBUG  [info] [NotificationService] tra-la-la - "" 
```

And then this line gets self replicated:

```
 DEBUG  [info] '[info] [NotificationService] tra-la-la - ""', '' - "" 
 DEBUG  [info] '[info] \'[info] [NotificationService] tra-la-la - ""\', \'\' - ""', '' - "" 
 DEBUG  [info] '[info] \'[info] \\\'[info] [NotificationService] tra-la-la - ""\\\', \\\'\\\' - ""\', \'\' - ""', '' - "" 
 DEBUG  [info] '[info] \'[info] \\\'[info] \\\\\\\'[info] [NotificationService] tra-la-la - ""\\\\\\\', \\\\\\\'\\\\\\\' - ""\\\', \\\'\\\' - ""\', \'\' - ""', '' - "" 
```

It happens only when I run app from XCode. So for me it looks like console output gets send to XCode and it produces an infinite loop.

Anyway, it's kind of rare use case (you need to run the app from XCode and receive a push notification to reproduce this), but I think it would be better to sanitize/modify logs on JS side to be sure that they are not getting recognized again by existing filter and don't produce infinite loop. What do you think?

Reply:
Alright let's try to work around that but still get the same end result. So let's leave the log statements in Swift as `[NotificationService] ...` and replace them with `[PushNotification] NotificationService - ...` in JS. So the new log message shouldn't match the filter anymore and avoid an infinite loop.

Reply:
Should be fixed now!